lib/port_scanner.py

```diff
@@ -15,25 +15,23 @@ def scan_port(host, port):
         sock.close()
 
 
-def scan_thread(host):
-    global queue
-    # TODO reimplement this loop with multiple ranges and threads
-    while True:
-        port_number = queue.get()
-        scan_port(host, port_number)
-        queue.task_done()
+def scan_thread(host, thread_start, thread_end):
+    for port in range(thread_start, thread_end + 1):
+        scan_port(host, port)
 
 
-def start_port_scan(host, ports, threads):
-    global queue
-    for thread in range(threads):
-        thread = Thread(target=scan_thread, args=(host,))
-        # if daemon is True that thread will end when the main threadends
-        thread.daemon = True
+def start_port_scan(host, ports_range, n_threads):
+    start, end = ports_range[0], ports_range[-1]    
```

Comment:
I actually wanted to comment previously that the port splitting should be done in the library not the cli code but I didn't do that because I saw opportunity to in the future, allow users to arbitrary select ports to scan, not only ranges. What you did here removes that possibility. But in the future we could add another function which would be used only for scanning arbitrary ports. Maybe change the function name to `scan_port_range`?

Also, the user is providing the ports in this format: `1-100`. Then you split this string and create a List: `[1, 2, 3, ..., 100]` just to grab the first and last element. Why not just accept the `start_port` and `end_port` in this function? This would also simplify code in `PixelToolkit.py`